### ISSUE TITLE
Can't seem to interrupt TTS with new wakeword detection

### ISSUE THREAD

**Issue by Leo-Stanoev** [rhasspy/wyoming-satellite#250]
Hi, I am trying to make it so that if it detects a new instance of the wake word it kills the ongoing tts message.

My satellite is configured as such right now, can't seem to get the right flags unfortunately.

[Unit]
Description=Wyoming Satellite
Wants=network-online.target
After=network-online.target
Requires=wyoming-openwakeword.service
Requires=2mic_leds.service
After=wyoming-openwakeword.service
After=2mic_leds.service
[Service]
Type=simple
#Environment="PULSE_SERVER=unix:/run/user/1000/pulse/native"
ExecStart=/home/leo/wyoming-satellite/script/run \
        --name 'Raspy Satelite' \
        --area 'Bedroom' \
        --uri 'tcp://0.0.0.0:10700' \
        --mic-command 'arecord -D plughw:CARD=seeed2micvoicec,DEV=0 -r 16000 -c 1 -f S16_LE -t raw' \
        --snd-command 'aplay -D plughw:CARD=seeed2micvoicec,DEV=0 -r 16000 -c 1 -f S16_LE -t raw' \
        --snd-command-rate 16000 \
        --snd-volume-multiplier 0.1 \
        --mic-auto-gain 2 \
        --mic-noise-suppression 3 \
        --mic-volume-multiplier 3 \
        --wake-uri 'tcp://127.0.0.1:10400' \
        --wake-word-name 'jarvis' \
        --detection-command 'examples/commands/detection_script.sh' \
        --wake-refractory-seconds 0.1 \
        --mic-no-mute-during-awake-wav \
        --event-uri 'tcp://127.0.0.1:10500'
WorkingDirectory=/home/leo/wyoming-satellite
Restart=always
User=leo
RestartSec=1

[Install]
WantedBy=default.target

This is my detection_script.sh

 #!/usr/bin/env sh

ps -A | grep aplay | awk '{print $1}' | xargs kill

It works if run manually via cli so am fairly sure it should work within the service, but maybe it doesn't run async wakeword detection if tts is streaming?

Any help would be appreciated to be able to interrupt the assistant if needed.

**Comment by mitrokun** [comment]
This is a very necessary feature, especially when start testing LLM.

**Comment by synesthesiam** [comment]
I agree, this is a good feature. We already have this for ESPHome voice satellites.

**Comment by mitrokun** [comment]
It would be great to use a second  OWW service with the  "stop" word to interrupt  to interrupt the current session.

**Comment by domesticatedviking** [comment]
- I would like to be able to use many different Assist agents concurrently, each of which would specialize in a particular type of interaction (local LLM queries, OpenAI, some which only use non-verbal responses, some purely to showcase all my custom Piper voices)   
- I would like to be able to interrupt the output.
- I would also like to be able to have natural multi-round conversations with my selected agent (ie without using the wake word every time)

Does anyone know how far Home Assistant is from officially incorporating any of these things?

It seems to me that accomplishing all of these things right now with official Home Assistant tools is probably not possible, that OpenWakeWord might not be the right tool for this job (other than for interrupting output) and that I might be better off to use my mic array's VAD capabilities to send all voice data it receives to a local whisper server, scan the resultant text for keywords configured on the satellite, and send them to the relevant Assist agent directly. 

Any thoughts?

**Comment by idjmic** [comment]
I'm after the same thing, been working on my satellite for past 2 weeks.
Lots of work, lots doesn't work.
